### ISSUE TITLE
macOS installer pkg fails with command-line install

### ISSUE THREAD

**Issue by gregneagle** [containers/podman#17910]
### Issue Description

TL; DR: attempts to upgrade Podman by installing the current release package on macOS fail.

Attempting to install the amd64-4.4.3 package for macOS from the command line fails, when run as root, or via sudo:
```
(as root)
# installer -pkg podman-installer-macos-amd64-4.4.3.pkg -target /
installer: Package name is Podman 4.4.3
installer: Upgrading at base path /
installer: The upgrade failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “podman-installer-macos-amd64-4.4.3.pkg”.)
```
```
(via sudo)
$ sudo installer -pkg podman-installer-macos-amd64-4.4.3.pkg -target /
installer: Package name is Podman 4.4.3
installer: Upgrading at base path /
installer: The upgrade failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “podman-installer-macos-amd64-4.4.3.pkg”.)
```

The issue is in the postinstall script:
```
#!/bin/bash

set -e

echo "/opt/podman/bin" > /etc/paths.d/podman-pkg

ln -s /opt/podman/bin/podman-mac-helper /opt/podman/qemu/bin/podman-mac-helper
ln -s /opt/podman/bin/gvproxy /opt/podman/qemu/bin/gvproxy

/opt/podman/bin/podman-mac-helper install
```

Specifically, `/opt/podman/bin/podman-mac-helper install`. This tool fails a variety of ways. First: if an older version of Podman is already installed:

```
bash-3.2# /opt/podman/bin/podman-mac-helper install
Error: helper is already installed, uninstall first
```
Instead of failing, it should either upgrade the tool, or do the uninstall/install dance. But just failing the entire install seems like  a bad idea.

I've also seen it fail with
```
Error: unexpected root user
```

### Steps to reproduce the issue

Steps to reproduce the issue
1. Install an older release of Podman
2. Download and install the current release: sudo installer -pkg podman-installer-macos-amd64-4.4.3.pkg -target /

### Describe the results you received

```
installer: The upgrade failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “podman-installer-macos-amd64-4.4.3.pkg”.)
```

### Describe the results you expected

Successful install

### podman info output

```yaml
podman 4.4.3, macOS, amd64 (x86_64)
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by Luap99** [comment]
Likely caused by https://github.com/containers/podman/pull/17786.
IMO If the helper is already installed it should not be treated as error in the podman-mac-helper binary.

For the unexpected root user I am not so sure, the auto install was added in https://github.com/containers/podman/pull/16565
The mac helper is no hard requirement so I assume it would be better to add `|| :` to make the install script ignore errors for it. This would at least restore the previous behaviour.

cc @anjannath @benoitf @n1hility

**Comment by benoitf** [comment]
yes if helper is already installed it should just continue

**Comment by anjannath** [comment]
maybe we should change it to always try to uninstall first with `/opt/podman/bin/podman-mac-helper install || true` and then install

**Comment by gregneagle** [comment]
If I completely remove podman (sudo rm -rf /opt/podman) and restart, and then attempt (as root) to install podman,
```
bash-3.2# installer -pkg podman-installer-macos-amd64-4.4.3.pkg -target /
installer: Package name is Podman 4.4.3
installer: Installing at base path /
installer: The install failed. (The Installer encountered an error that caused the installation to fail. Contact the software manufacturer for assistance. An error occurred while running scripts from the package “podman-installer-macos-amd64-4.4.3.pkg”.)
```

Snippet from /var/log/install.log:
```
2023-03-24 09:30:32-07 arethafranklin installd[1531]: PackageKit (package_script_service): Preparing to execute script "./postinstall" in /private/tmp/PKInstallSandbox.TXcjQ9/Scripts/com.redhat.podman.94dOQD
2023-03-24 09:30:32-07 arethafranklin package_script_service[2019]: PackageKit: Executing script "postinstall" in /tmp/PKInstallSandbox.TXcjQ9/Scripts/com.redhat.podman.94dOQD
2023-03-24 09:30:32-07 arethafranklin package_script_service[2019]: Set responsibility to pid: 2853, responsible_path: /usr/sbin/installer
2023-03-24 09:30:32-07 arethafranklin package_script_service[2019]: ./postinstall: Error: unexpected root user
```
This is `podman-mac-helper` failing with 
```
Error: unexpected root user
```

Would you like this opened as a separate issue, or are you happy with just ignoring the error as you do here: https://github.com/containers/podman/pull/17916

**Comment by Luap99** [comment]
you need to run it with sudo AFAICT so it can get your user id from the sudo env vars

**Comment by gregneagle** [comment]
That's not helpful in a mass-deployment scenario. Software is not installed interactively by users -- we have a software distribution tool that runs as root. (In our case, we use Munki, but this would be the case with _any_ software management tool: Jamf, Addigy, VMware Workspace ONE, SimpleMDM, FileWave, etc, or even configuration management tools like Ansible, Puppet, Chef, or SaltStack)

**Comment by benoitf** [comment]
will it be included in an 4.4.4 or an updated 4.4.3 .PKG packages ?

**Comment by gregneagle** [comment]
I'd really hope it would be in a 4.4.4 release, or a 4.4.3.x release. It's difficult manage packages that are different, but have the same version number.

**Comment by Luap99** [comment]
see the discussion on the PR, goal is to create a 4.4.4 version

**Comment by n1hility** [comment]
@LtCmdrScott  The install process needs to know the user to install the command with, so if it cant determine this it will fail.  "Getting Error: unexpected root user" can happen if the user running the sudo command was root, and SUDO_USER wasn't defined (normally defined by sudo to the calling user [or mirrored to allow nesting]). One way this can occur is if the environment was dropped via some other nested call to other user invoking tools (e.g. su ).
